### ISSUE TITLE
Support yyaxis

### ISSUE THREAD

**Issue by egeerardyn** [matlab2tikz/matlab2tikz#984]
In MATLAB, the function `plotyy` is being phased out in more recent releases, e.g. as noted on FileExchange:

>This package is great, thank you very much for all the work involved!
>
> Recently, MATLAB changed their syntax for double axis plots, and the command plotyy() is not recommended anymore. When using yyaxis() instead (which is what they recommend) the figure doesn't have two axes objects anymore, but just one. 
> While matlab2tikz worked fine when there were two axes objects (handles of which could be retrieved with ah = findobj(gcf,'Type','Axes') ), it doesn't display the left-side plot any more in the new format. Could you possibly fix this? That would be of great help to me, thank you!

We already had a issue #905 around `yyaxis`, but we should support the functionality officially as well.

**Comment by HannesThielke** [comment]
hey everyone,
this issue was opened a few years ago but it seems that it is still open.
I am facing problems with the workaround with plotyy provided in #905.
Is there a way to use yyaxis with Matlab2tikz since plotyy is not recommended by Matlab anymore.
Best wishes,
Hannes

**Comment by ProcyonTheEvil** [comment]
With yyaxis, one has to do the following:

1. Call yyaxis(ax, 'left')
2. Read all Children of the left y-Axis: ax.Children
3. Call yyaxis(ax, 'right')
4. Read all Children of the right y-Axis: ax.Children

(Do the same with ax.XAxisLocation = 'bottom' or 'top'. ax.ZAxisLocation does not exist in Matlab R2020b)

If you only use the allchild function, you get all the matlab.graphics.chart.primitive.Line elements, but no indication whether they belong to the left or right y axis.

**Comment by wil2810** [comment]
I have the same problem. I'm using `yyaxis` (with Matlab R2020a).
 
@ProcyonTheEvil, if I understand you correctly, this is a workaround to use yyaxis with matlab2tikz. Can you explain this in more detail? 
I tried what you suggested in steps 1-4, but with no success. Here is what I did:
```matlab
% first create figure; then:
ax = gca;
yyaxis(ax, 'left')
ax.Children;
yyaxis(ax, 'right')
ax.Children;
matlab2tikz('myFig.tex', 'extraAxisOptions', 'scale=\figurescale');
```
What am I missing?

Thank you!

**Comment by ProcyonTheEvil** [comment]
@wil2810
No, this is not a workaround. This is how matlab2tikz code could be changed in order to get a list of all plot lines and their corresponding y axes. Matlab2tikz cannot handle two y axes right now.

Am 11. Mai 2021 16:52:17 MESZ schrieb Tim Wagner ***@***.***>:
>I have the same problem. I'm using `yyaxis` (with Matlab R2020a).
> 
>@ProcyonTheEvil, if I understand you correctly, this is a workaround to
>use yyaxis with matlab2tikz. Can you explain this in more detail? 
>I tried what you suggested in steps 1-4, but with no success. Here is
>what I did:
>```matlab
>% first create figure; then:
>ax = gca;
>yyaxis(ax, 'left')
>ax.Children;
>yyaxis(ax, 'right')
>ax.Children;
>matlab2tikz('myFig.tex', 'extraAxisOptions', 'scale=\figurescale');
>```
>What am I missing?
>
>Thank you!
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/matlab2tikz/matlab2tikz/issues/984#issuecomment-838615948

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by wil2810** [comment]
@ProcyonTheEvil 
Oh, thanks for the clarification! I completely misunderstood you before. Sorry.
Also thanks for the quick answer. :)
